Adding cosi specific changes to prow.sh

brahmaroutu opened the pull request:
prow.sh is modifies to use COSI specific config variables. It installs crds, controller and has hooks to run serial and parallel e2e tests.

brahmaroutu commented:
/assign @xing-yang

xing-yang commented on release-tools/prow.sh:
```diff
@@ -291,72 +212,24 @@ tests_need_alpha_cluster () {
-configvar CSI_PROW_E2E_FOCUS_1_16 '^' "non-alpha, feature-tagged tests for Kubernetes = 1.16" # no tests to run, match nothing
-configvar CSI_PROW_E2E_FOCUS_LATEST '\[Feature:VolumeSnapshotDataSource\]' "non-alpha, feature-tagged tests for Kubernetes >= 1.17"
-configvar CSI_PROW_E2E_FOCUS "$(get_versioned_variable CSI_PROW_E2E_FOCUS "${csi_prow_kubernetes_version_suffix}")" "non-alpha, feature-tagged tests"
+configvar COSI_PROW_E2E_FOCUS "$(get_versioned_variable COSI_PROW_E2E_FOCUS "${cosi_prow_kubernetes_version_suffix}")" "non-alpha, feature-tagged tests"
```
I'm not sure if the tag is relevant for COSI.  For CSI tests, tests tagged with feature gates are only run in clusters that support them.  We can keep them for now and remove them if we are sure they are not needed.

brahmaroutu replied:
Fixed.

xing-yang commented on release-tools/prow.sh:
```diff
@@ -652,103 +515,74 @@ find_deployment () {
+# Installs all nessesary CRDs  
+install_crds() {
+  # Wait until cosi CRDs are in place.
+#  CRD_BASE_DIR="https://github.com/kubernetes-sigs/container-object-storage-interface-api/crds"
```
Remove this

brahmaroutu replied:
Done

xing-yang commented on release-tools/prow.sh:
```diff
@@ -652,103 +515,74 @@ find_deployment () {
+install_crds() {
+  # Wait until cosi CRDs are in place.
+#  CRD_BASE_DIR="https://github.com/kubernetes-sigs/container-object-storage-interface-api/crds"
+  CRD_BASE_DIR="https://raw.githubusercontent.com/brahmaroutu/api/proper_crds/crds"
```
This should be from an official repo?
If you are still testing, can you mark your PR as "WIP"?

brahmaroutu replied:
Fixed it to use the official site (now available). But I need to work with a different error, 
"Required value: protected groups must have approval annotation "api-approved.kubernetes.io", see https://github.com/kubernetes/enhancements/pull/1111"
I am implementing a workaround using sed to insert the annotation until we move these api group into k8s build.

xing-yang replied:
I think you can use this PR "https://github.com/kubernetes-sigs/container-object-storage-interface-api/pull/2" for now and replace it with the approved PR in the future when it is officially approved by the API reviewer.

brahmaroutu replied:
Fixed

xing-yang commented on release-tools/prow.sh:
```diff
@@ -652,103 +515,74 @@ find_deployment () {
+  # Wait until cosi CRDs are in place.
+#  CRD_BASE_DIR="https://github.com/kubernetes-sigs/container-object-storage-interface-api/crds"
+  CRD_BASE_DIR="https://raw.githubusercontent.com/brahmaroutu/api/proper_crds/crds"
+  kubectl apply -f "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketclasses.yaml" --validate=false
```
The install location should be dynamic. If this is testing a PR at the API repo "https://github.com/kubernetes-sigs/container-object-storage-interface-api", it should read from the  API repo.  In other cases, it should either install from master or a stable branch of the API repo.

See this PR for example: https://github.com/kubernetes-csi/csi-release-tools/pull/113

brahmaroutu replied:
Fixed

xing-yang commented on release-tools/prow.sh:
```diff
@@ -652,103 +515,74 @@ find_deployment () {
+# Install controller and associated RBAC, retrying until the pod is running.
+install_controller() {
+  #kubectl apply -f "https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-controller/${COSI_CONTROLLER_VERSION}/deploy/controller.yaml"
+  kubectl apply -f "https://raw.githubusercontent.com/container-object-storage-interface/cosi-controller-manager/master/samples/rbac.yaml"
```
See my comments earlier about install location should be dynamic.

brahmaroutu replied:
Fixed

xing-yang commented on release-tools/prow.sh:
```diff
@@ -652,103 +515,74 @@ find_deployment () {
 
-  kubectl apply -f "https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/${CSI_SNAPSHOTTER_VERSION}/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml"
+  #kubectl apply -f "https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-controller/${COSI_CONTROLLER_VERSION}/deploy/rbac.yaml"
+  kubectl apply -f "https://raw.githubusercontent.com/brahmaroutu/cosi-controller-manager/test/samples/controller.yaml"
```
See my comments earlier about install location should be dynamic.

brahmaroutu replied:
Fixed

xing-yang commented on release-tools/prow.sh:
```diff
@@ -770,9 +604,9 @@ $(
-done
+#docker exec cosi-prow-control-plane docker image ls --format='{{.Repository}} {{.Tag}} {{.ID}}' | grep -e COSI -e hostpath | while read -r repo tag id; do
+#    echo "$repo" "$tag" "$(docker exec cosi-prow-control-plane docker image inspect --format='{{ index .Config.Labels "revision"}}' "$id")"
+#done
```
Remove if no longer needed.

brahmaroutu replied:
done

xing-yang commented on release-tools/prow.sh:
```diff
@@ -1077,106 +852,49 @@ main () {
-                    fi
+            echo "Srini Collect Cluster Info"
+            collect_cluster_info 
+            echo "Srini Done Collect Cluster Info"
```
Looks like these are your temporary debug messages.
Can you mark this PR as WIP?  Looks like you are still testing.

brahmaroutu replied:
Fixed.

NicolasT commented on release-tools/prow.sh:
```diff
@@ -652,103 +511,78 @@ find_deployment () {
+# Installs all nessesary CRDs  
+install_crds() {
+  # Wait until cosi CRDs are in place.
+  CRD_BASE_DIR="https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketclasses.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketrequests.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_buckets.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketaccessclasses.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketaccessrequests.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketaccesses.yaml"
```
Since https://github.com/kubernetes-sigs/container-object-storage-interface-api/pull/13 there's a `kustomization.yaml` in https://github.com/kubernetes-sigs/container-object-storage-interface-api/tree/master/crds wich allows to deploy the CRDs (including required `labels`) without exhaustively listing all of them here.

You could use, as an example, `kustomize build "https://github.com/kubernetes-sigs/container-object-storage-interface-api/crds/?ref=master" | kubectl apply -f -`.

brahmaroutu commented:
@xing-yang please review, I have addressed your review comments. This runs fine but may require further changes as we introduce post submit jobs, e2e/integration testing and other cost components into the build

xing-yang commented on release-tools/prow.sh:
```diff
@@ -1036,32 +805,32 @@ main () {
             for i in $cmds; do
                 e=$(echo "$i" | tr '[:lower:]' '[:upper:]' | tr - _)
-                images="$images ${e}_REGISTRY=none ${e}_TAG=csiprow"
+                images="$images ${e}_REGISTRY=quay.io/containerobjectstorage ${e}_TAG=cosiprow"
```
Are you going to use quay.io to store images, not k8s.gcr.io?

xing-yang replied:
Discussed at the review meeting.  Will use quay.io for now and move to k8s.gcr.io in the future.

xing-yang commented:
/lgtm
/approve
